src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+        require(msg.sender == owner || msg.sender == pollAuthor[_pollNum], "UNAUTHORIZED");
+        _;
+    }
+
+
+    // ------
+    // Events
+    // ------
+
+    /// @notice Emitted when a new poll is created.
+    event ProposalCreated(uint256 pollNum, PollType pollType, uint256 endTime);
+    /// @notice Emitted when a poll has been passed.
+    event ProposalPassed(uint256 pollNum);
+    /// @notice Emitted when the status of gateKeeping is updated.
+    event GateKeepingModified(bool enabled);
+
+
+    // ---------
+    // Functions
+    // ---------
+
+    /// @notice Utility function so this contract can accept BNB.
     receive() payable external {}
 
-    // ---------- Votes ----------
+    /// @notice is used to create a new poll.
+    /// @param  _pollType enum type of poll being created.
+    /// @param  _change the matching metadata that will result in the execution of the poll.
+    function createPoll(PollType _pollType, Metadata memory _change) public {        
+        require(createPollEnabled, "GogeDao.sol::createPoll() Ability to create poll is disabled");
+        if (msg.sender != owner) require(getActivePollsFromAuthor(msg.sender) < maxPollsPerAuthor, "GogeDao.sol::createPoll() Exceeds maxPollsPerAuthor");
+        require(block.timestamp < _change.endTime, "GogeDao.sol::createPoll() End time must be later than start time");
+        require(_change.endTime - block.timestamp >= minPeriod, "GogeDao.sol::createPoll() Polling period must be greater than 24 hours");
+
+        require(IGogeERC20(governanceTokenAddr).balanceOf(msg.sender) >= minAuthorBal, "GogeDao.sol::createPoll() Insufficient balance of tokens");
+        require(IGogeERC20(governanceTokenAddr).transferFrom(msg.sender, address(this), minAuthorBal));
+
+        emit ProposalCreated(pollNum, _pollType, _change.endTime);
 
-    /// @notice A method for a voter to create a vote.
+        pollNum += 1;
+
+        _addToVoterLibrary(pollNum, msg.sender);
+        _addToAdvocateFor(pollNum, msg.sender);
+
+        polls[pollNum][msg.sender] += minAuthorBal;
+        totalVotes[pollNum]        += minAuthorBal;
+
+        pollTypes[pollNum]     = _pollType;
+        pollMap[pollNum]       = _change;
+        pollStartTime[pollNum] = block.timestamp;
+        pollEndTime[pollNum]   = _change.endTime;
+        pollAuthor[pollNum]    = msg.sender;
+
+        activePolls.push(pollNum);
+    }
+
+    /// @notice A method for a voter to add a vote to an existing poll.
     /// @param  _pollNum The poll number.
     /// @param  _numVotes The size of the vote to be created.
     function addVote(uint256 _pollNum, uint256 _numVotes) public {
+        require(_pollNum <= pollNum, "GogeDao.sol::addVote() Poll doesn't Exist");
+        require(block.timestamp >= pollStartTime[_pollNum] && block.timestamp < pollEndTime[_pollNum], "GogeDao.sol::addVote() Poll Closed");
+        require(isActivePoll(_pollNum), "GogeDao.sol::addVote() Poll is not active");
```

Comment:
Yeah I think checking for _pollNum to be less than pollNum is not necessary with the introduction of activePolls and we check if the poll is active 2 lines later. We also dont need the check to see if the poll is closed since we have queryEndTime. However, In the off chance that queryEndTime was not called due to some issue; polls that are expired will still exist in activePolls so do we want people to still be able to vote on expired polls?